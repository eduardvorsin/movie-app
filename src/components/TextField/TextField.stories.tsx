import { Meta, StoryObj } from "@storybook/react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { action } from '@storybook/addon-actions';
import TextField, { Props } from "./TextField";
import { ChangeEventHandler, useState } from "react";

const meta: Meta<typeof TextField> = {
	title: 'components/TextField',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for the TextField'
		},
		testId: {
			description: 'Id for testing the component',
		},
		label: {
			description: 'Text description for the input field',
		},
		onClear: {
			control: false,
			description: 'The callback is performed by pressing the clear button'
		},
		isDisabled: {
			description: 'Disabling input makes it non-functional',
		},
		isInvalid: {
			description: 'Visually highlights the input in red when incorrectly entered data',
		},
		isReadOnly: {
			description: 'Makes the input read-only',
		},
		isRequired: {
			description: 'Makes input mandatory and adds a red asterisk at the end of the label',
		},
		name: {
			description: 'Name for the input',
		},
		placeholder: {
			description: 'The filling text that is displayed when the input is empty',
		},
		onChange: {
			description: 'The callback is executed when the input value changes',
			control: false,
		},
		value: {
			description: 'Input value',
		},
		labelHidden: {
			description: 'Makes the label visible only to screen readers',
		},
		clearButton: {
			description: 'Allows you to show or hide the input value clearing button',
		},
		error: {
			description: 'Error Message',
		},
		id: {
			description: 'Component id',
		},
		autoComplete: {
			description: 'Allows you to set the type of autofill',
		},
		pattern: {
			description: 'Allows you to set a regular expression pattern for validation',
		},
		maxLength: {
			description: 'Allows you to set the maximum length of the value',
		},
		minLength: {
			description: 'Allows you to set the minimum length of the value',
		},
		onFocus: {
			description: 'The callback is executed when the focus hits the input',
			control: false,
		},
		onBlur: {
			description: 'The callback is executed when the input loses focus',
			control: false,
		},
		inputMode: {
			description: 'Allows you to set the keyboard type on mobile devices when entering a value',
		},
		type: {
			description: 'Input type',
		}
	},
	component: TextField,
	parameters: {
		docs: {
			description: {
				component: 'The input component is used to receive data or information from the user and then transmit it.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof TextField>;

const TextFieldWithHooks = (props: Omit<Props, 'onChange' | 'value'>) => {
	const [value, setValue] = useState<string>('');

	const changeHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
		setValue(e.currentTarget.value);
		action('Changed')(e);
	}

	const clearHandler = () => {
		if (props.onClear) props.onClear();
		setValue('');
	}

	return (
		<TextField
			{...props}
			value={value}
			onClear={clearHandler}
			onChange={changeHandler}
		/>
	)
};

export const Default: Story = {
	args: {
		label: 'Default',
		name: 'Default',
		id: 'Default',
	},
	parameters: {
		docs: {
			description: {
				story: 'Default state of the input',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />),
};

export const Disabled: Story = {
	args: {
		label: 'Disabled',
		name: 'Disabled',
		id: 'Disabled',
		isDisabled: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Apply this to actions that are currently unavailable. The user interface should effectively convey why the input is not active and what steps are needed to activate it.',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const Invalid: Story = {
	args: {
		label: 'Invalid',
		name: 'Invalid',
		id: 'Invalid',
		isInvalid: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to visually highlight the input in red when it does not pass validation.',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const Required: Story = {
	args: {
		label: 'Required',
		name: 'Required',
		id: 'Required',
		isRequired: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to indicate the input must be filled in.',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const Placeholder: Story = {
	args: {
		label: 'Placeholder',
		name: 'Placeholder',
		id: 'Placeholder',
		placeholder: 'Placeholder',
	},
	parameters: {
		docs: {
			description: {
				story: 'Used to show an example of a value that can be entered into an input',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const ClearButton: Story = {
	args: {
		label: 'ClearButton',
		name: 'ClearButton',
		id: 'ClearButton',
		clearButton: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Shows a button when clicked on which the contents of the input are cleared',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const EventCallbacks: Story = {
	argTypes: {
		onClear: {
			action: 'Cleared',
		},
		onFocus: {
			action: 'Focused',
		},
		onBlur: {
			action: 'Blured',
		},
	},
	args: {
		name: 'event callbacks',
		id: 'event callbacks',
		label: 'clear, focus, blur event',
		clearButton: true,
		value: '',
	},
	parameters: {
		docs: {
			description: {
				story: 'The callback function can be provided to the onFocus, onBlur event and when the clear button is clicked',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const LabelHidden: Story = {
	args: {
		label: 'LabelHidden',
		name: 'LabelHidden',
		id: 'LabelHidden',
		labelHidden: true,
	},
	parameters: {
		docs: {
			description: {
				story: 'Makes the label visible only to screen readers',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

export const Error: Story = {
	args: {
		label: 'Error',
		name: 'Error',
		id: 'Error',
		error: 'error messsage',
	},
	parameters: {
		docs: {
			description: {
				story: 'Displays an error message under the input',
			},
		},
	},
	render: (args) => (<TextFieldWithHooks {...args} />)
};

type Variant = {
	label: Props['label']
	isDisabled?: boolean,
	isInvalid?: boolean,
	isReadOnly?: boolean,
	isRequired?: boolean,
	labelHidden?: boolean,
	clearButton?: boolean,
	error?: Props['error'],
	placeholder?: Props['placeholder'],
}

const variants: Variant[] = [
	{
		label: 'disabled',
		isDisabled: true,
	},
	{
		label: 'invalid',
		isInvalid: true,
	},
	{
		label: 'read only',
		isReadOnly: true,
	},
	{
		label: 'required',
		isRequired: true,
	},
	{
		label: 'label hidden',
		labelHidden: true,
	},
	{
		label: 'clear button',
		clearButton: true,
	},
	{
		label: 'error',
		error: 'error',
	},
	{
		label: 'placeholder',
		placeholder: 'placeholder',
	},
]

export const All: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Here are all possible variants of inputs.',
			},
		},
	},
	render: () => (
		<div
			className='grid gap-8 grid-cols-4 grid-rows-2'
		>
			{variants.map((variant) => (
				<TextFieldWithHooks
					key={variant.label}
					id={variant.label}
					name={variant.label}
					label={variant.label}
					isDisabled={variant.isDisabled}
					isInvalid={variant.isInvalid}
					isReadOnly={variant.isReadOnly}
					isRequired={variant.isRequired}
					labelHidden={variant.labelHidden}
					clearButton={variant.clearButton}
					error={variant.error}
					placeholder={variant.placeholder}
				/>
			))}
		</div>
	),
};