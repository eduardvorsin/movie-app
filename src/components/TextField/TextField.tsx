'use client';
import { ChangeEventHandler, FocusEventHandler } from 'react';
import InlineMessage from '@/components/InlineMessage/InlineMessage';
import { GeneralProps } from '@/types/shared';
import { useParams } from 'next/navigation';
import { Locales, fallbackLng } from '@/i18n/settings';
import { useTranslation } from '@/i18n/client';

export type Props = {
	isDisabled?: boolean,
	isInvalid?: boolean,
	isReadOnly?: boolean,
	isRequired?: boolean,
	name: string,
	placeholder?: string,
	onChange: ChangeEventHandler<HTMLInputElement>,
	value: string,
	label: string,
	labelHidden?: boolean,
	clearButton?: boolean,
	error?: string,
	id: string,
	autoComplete?: string,
	pattern?: string,
	maxLength?: number,
	minLength?: number,
	onClear?: () => void,
	onFocus?: FocusEventHandler<HTMLInputElement>,
	onBlur?: FocusEventHandler<HTMLInputElement>,
	inputMode?: 'none' | 'text' | 'decimal' | 'numeric' | 'tel' | 'search' | 'email' | 'url'
	type?: 'text' | 'email' | 'password' | 'search' | 'tel' | 'url'
} & GeneralProps;

export default function TextField({
	className,
	isDisabled,
	isInvalid,
	isReadOnly,
	isRequired,
	name,
	placeholder,
	onChange,
	testId,
	value,
	label,
	labelHidden,
	clearButton,
	error,
	id,
	autoComplete,
	pattern,
	maxLength,
	minLength,
	onClear,
	onFocus,
	onBlur,
	inputMode,
	type,
	...props
}: Props) {
	const lang = useParams()?.lang as Locales ?? fallbackLng;
	const { t } = useTranslation(lang);

	const labelClasses = [
		'block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150',
		labelHidden ? 'sr-only' : ''
	].join(' ');
	const inputClasses = [
		'cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300',
		isInvalid ? 'border-red-600 dark:border-red-300' : 'border-neutral-300 dark:border-dark-neutral-350',
		clearButton ? 'pl-3 pr-8' : 'px-3',
		className
	].join(' ');

	return (
		<div
			className={isDisabled ? 'opacity-disabled cursor-not-allowed' : ''}
			data-testid={testId}
			{...props}
		>
			<label
				className={labelClasses}
				id={`label-${id}`}
				htmlFor={id}
			>
				{label}
				{isRequired && (
					<span
						className='pl-0.5 text-red-800 dark:text-red-300 transition-colors duration-150'
						aria-hidden='true'
					>
						*
					</span>
				)}
			</label>
			<div
				className='relative'
			>
				<input
					className={inputClasses}
					value={value}
					id={id}
					name={name}
					role={type === 'search' ? 'searchbox' : undefined}
					disabled={isDisabled}
					required={isRequired}
					type={type === 'search' ? 'text' : type}
					inputMode={inputMode}
					minLength={minLength}
					maxLength={maxLength}
					pattern={pattern}
					autoComplete={autoComplete}
					placeholder={placeholder}
					readOnly={isReadOnly}
					onChange={onChange}
					onFocus={onFocus}
					onBlur={onBlur}
					aria-labelledby={error ? `${id}-error-message` : undefined}
					aria-invalid={isInvalid}
				/>
				{clearButton && (
					<button
						className='text-dark-neutral-0 dark:text-neutral-400 w-5 h-5 absolute top-1/2 right-[0.5rem] -translate-y-1/2 text-0 transition-colors duration-150'
						onClick={onClear}
					>
						{t('textField.button')}
						<svg className='fill-current' viewBox='0 0 20 20'>
							<use href={'/assets/icons/cancel.svg#cancel'}></use>
						</svg>
					</button>
				)}
			</div>
			{error && (
				<InlineMessage
					className='mt-1'
					message={error}
					fieldId={`${id}-error-message`}
				/>
			)}
		</div>
	);
};