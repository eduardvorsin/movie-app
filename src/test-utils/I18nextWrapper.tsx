import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;