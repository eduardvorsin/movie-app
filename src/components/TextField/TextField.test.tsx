import { render, screen } from '@testing-library/react';
import TextField from './TextField';
import userEvent from '@testing-library/user-event';
import i18next from '@/i18n/client';
import I18nextWrapper from '@/test-utils/I18nextWrapper';

describe('TextField tests', () => {
	it('is rendered correctly', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toBeInTheDocument();
	});

	it('the mock function is triggered when a value is entered into the input', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				clearButton={false}
			/>
		);

		await user.type(screen.getByRole<HTMLInputElement>('textbox'), 'abcdef');

		expect(mockFn).toHaveBeenCalledTimes(6);
	})

	it('when the isRequired prop is true, an asterisk appears at the end of the label', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				isRequired
				clearButton={false}
			/>
		);

		expect(screen.getByText<HTMLSpanElement>('*')).toBeInTheDocument();
	});

	it('when the prop IsInvalid is true the border color turns red', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				isInvalid
				clearButton={false}
			/>
		);

		expect(screen.getByRole<HTMLInputElement>('textbox')).toHaveClass('border-red-600');
	});

	it('The mock function is triggered when the focus hits the input', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				onFocus={mockFn}
				testId='test-field'
				clearButton={false}
			/>
		);

		await user.tab();

		expect(mockFn).toHaveBeenCalledTimes(1);
	})

	it('The mock function is triggered when the focus is lost from the input', async () => {
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				onBlur={mockFn}
				testId='test-field'
				clearButton={false}
			/>
		);

		await user.tab();
		await user.tab();

		expect(mockFn).toHaveBeenCalledTimes(1);
	});

	it('the mock function is triggered when the clear button is clicked', async () => {
		const dictionary = { clearButton: 'To clear' };
		const user = userEvent.setup();
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				onClear={mockFn}
				clearButton
				dictionary={dictionary}
			/>
		);

		await user.click(screen.getByRole<HTMLButtonElement>('button', { name: /clear/i }));

		expect(mockFn).toHaveBeenCalledTimes(1);
	});

	it('when the clearButton prop is true, the input clearing button appears', () => {
		const dictionary = { clearButton: 'To clear' };
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				clearButton
				dictionary={dictionary}
			/>
		);

		expect(screen.getByRole<HTMLButtonElement>('button', { name: /clear/i })).toBeInTheDocument();
	});

	it('when the labelHidden prop is true, the label becomes visible only to screen readers', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				labelHidden
				clearButton={false}
			/>
		);

		expect(screen.getByText<HTMLLabelElement>('test-field')).toHaveClass('sr-only');
	});

	it('when the error prop is not empty, an error message is displayed next to the input', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				error='mock error'
				clearButton={false}
			/>
		);

		expect(screen.getByText<HTMLParagraphElement>('mock error')).toBeInTheDocument();
	});

	it('is a basic snapshot', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with isDisabled set to true', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				isDisabled
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with isReadonly set to true', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				isReadOnly
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with isInvalid set to true', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				isInvalid
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with isRequired set to true', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				isRequired
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with clear button', () => {
		const dictionary = { clearButton: 'To clear' };
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				clearButton
				dictionary={dictionary}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with an error message', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				error='mock error'
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});

	it('is a snapshot with labelHidden set to true', () => {
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				labelHidden
				clearButton={false}
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-field')).toMatchSnapshot();
	});
});

describe('TextField integration tests', () => {
	it('localization into English works correctly', async () => {
		await i18next.changeLanguage('en');
		const translation = new RegExp(i18next.t('textField.clearButton'));

		const dictionary = { clearButton: i18next.t('textField.clearButton') };
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				labelHidden
				clearButton
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLSpanElement>(translation)).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		await i18next.changeLanguage('ru');
		const translation = new RegExp(i18next.t('textField.clearButton'));

		const dictionary = { clearButton: i18next.t('textField.clearButton') };
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				labelHidden
				clearButton
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLSpanElement>(translation)).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		await i18next.changeLanguage('en');

		const dictionary = { clearButton: i18next.t('textField.clearButton') };
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				labelHidden
				clearButton
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLDivElement>('test-field')).toBeInTheDocument();
	});

	it('is a snapshot with Russian localization', async () => {
		await i18next.changeLanguage('ru');

		const dictionary = { clearButton: i18next.t('textField.clearButton') };
		const mockFn = jest.fn();
		render(
			<TextField
				onChange={mockFn}
				name='test-field'
				label='test-field'
				id='test-field'
				value='mock'
				testId='test-field'
				labelHidden
				clearButton
				dictionary={dictionary}
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLDivElement>('test-field')).toBeInTheDocument();
	});
});