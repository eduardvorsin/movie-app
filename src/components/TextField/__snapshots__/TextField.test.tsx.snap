// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField tests is a basic snapshot 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 px-3 "
      id="test-field"
      name="test-field"
      value="mock"
    />
  </div>
</div>
`;

exports[`TextField tests is a snapshot with an error message 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      aria-labelledby="test-field-error-message"
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 px-3 "
      id="test-field"
      name="test-field"
      value="mock"
    />
  </div>
  <p
    class="flex items-center text-75 transition-colors duration-150 text-blue-700 dark:text-blue-400 mt-1"
    id="test-field-error-message"
    role="alert"
  >
    <span
      aria-hidden="true"
      class="mr-1 w-6 h-6 transition-colors duration-150"
    >
      <svg
        class="fill-current transition-colors duration-150"
        viewBox="0 0 20 20"
      >
        <use
          href="/assets/icons/connectivity.svg#connectivity"
        />
      </svg>
    </span>
    mock error
  </p>
</div>
`;

exports[`TextField tests is a snapshot with clear button 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 pl-3 pr-8 "
      id="test-field"
      name="test-field"
      value="mock"
    />
    <button
      class="text-dark-neutral-0 dark:text-neutral-400 w-5 h-5 absolute top-1/2 right-[0.5rem] -translate-y-1/2 text-0 transition-colors duration-150"
      type="button"
    >
      To clear
      <svg
        class="fill-current"
        viewBox="0 0 20 20"
      >
        <use
          href="/assets/icons/cancel.svg#cancel"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`TextField tests is a snapshot with isDisabled set to true 1`] = `
<div
  class="opacity-disabled cursor-not-allowed"
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 px-3 "
      disabled=""
      id="test-field"
      name="test-field"
      value="mock"
    />
  </div>
</div>
`;

exports[`TextField tests is a snapshot with isInvalid set to true 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      aria-invalid="true"
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-red-600 dark:border-red-300 px-3 "
      id="test-field"
      name="test-field"
      value="mock"
    />
  </div>
</div>
`;

exports[`TextField tests is a snapshot with isReadonly set to true 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 px-3 "
      id="test-field"
      name="test-field"
      readonly=""
      value="mock"
    />
  </div>
</div>
`;

exports[`TextField tests is a snapshot with isRequired set to true 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 "
    for="test-field"
    id="label-test-field"
  >
    test-field
    <span
      aria-hidden="true"
      class="pl-0.5 text-red-800 dark:text-red-300 transition-colors duration-150"
    >
      *
    </span>
  </label>
  <div
    class="relative"
  >
    <input
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 px-3 "
      id="test-field"
      name="test-field"
      required=""
      value="mock"
    />
  </div>
</div>
`;

exports[`TextField tests is a snapshot with labelHidden set to true 1`] = `
<div
  class=""
  data-testid="test-field"
>
  <label
    class="block mb-1 text-100 font-regular text-dark-neutral-0 dark:text-neutral-400 cursor-[inherit] transition-colors duration-150 sr-only"
    for="test-field"
    id="label-test-field"
  >
    test-field
  </label>
  <div
    class="relative"
  >
    <input
      class="cursor-[inherit] py-1 text-100 leading-2 font-regular block w-full min-h-[2.25rem] border-2 text-dark-neutral-0 placeholder-dark-neutral-700 rounded-1 transition-colors ease-ease-in-out duration-150 enabled:hover:bg-neutral-100 focus:outline-none enabled:focus:border-blue-500 dark:bg-dark-neutral-200 dark:enabled:hover:bg-dark-neutral-250 dark:text-neutral-400 dark:placeholder-neutral-600 dark:enabled:focus:border-blue-300 border-neutral-300 dark:border-dark-neutral-350 px-3 "
      id="test-field"
      name="test-field"
      value="mock"
    />
  </div>
</div>
`;
